hivemq-edge/src/frontend/src/extensions/datahub/flow/transition/TransitionPanel.tsx

```diff
@@ -1,29 +1,108 @@
-import { FC, useMemo } from 'react'
-import { Node } from 'reactflow'
+import { FC, useCallback, useMemo } from 'react'
+import { getIncomers, Node } from 'reactflow'
 import { Card, CardBody } from '@chakra-ui/react'
+import { IChangeEvent } from '@rjsf/core'
 
-import { PanelProps, TransitionData } from '@datahub/types.ts'
+import {
+  BehaviorPolicyData,
+  DataHubNodeType,
+  FiniteStateMachineSchema,
+  PanelProps,
+  StateType,
+  TransitionData,
+  TransitionType,
+} from '@datahub/types.ts'
 import useDataHubDraftStore from '@datahub/hooks/useDataHubDraftStore.ts'
 import { ReactFlowSchemaForm } from '@datahub/components/forms/ReactFlowSchemaForm.tsx'
 import { MOCK_TRANSITION_SCHEMA } from '@datahub/flow/transition/TransitionData.ts'
+import { datahubRJSFWidgets } from '@datahub/flow/datahubRJSFWidgets.tsx'
+import { MOCK_BEHAVIOR_POLICY_SCHEMA } from '@datahub/flow/behavior_policy/BehaviorPolicySchema.ts'
+import { FiniteStateMachineFlow } from '@datahub/components/fsm/FiniteStateMachineFlow.tsx'
 
 export const TransitionPanel: FC<PanelProps> = ({ selectedNode, onFormSubmit }) => {
-  const { nodes } = useDataHubDraftStore()
+  const { nodes, edges } = useDataHubDraftStore()
+
+  // TODO[NVL] Error messages?
+  const parentPolicy = useMemo(() => {
+    const adapterNode = nodes.find((e) => e.id === selectedNode) as Node<TransitionData> | undefined
+    if (!adapterNode) return undefined
+    const incomers = getIncomers(adapterNode, nodes, edges)
+    if (!incomers) return undefined
+    if (incomers.length !== 1) return undefined
+    const [behavior] = incomers
+    if (behavior.type !== DataHubNodeType.BEHAVIOR_POLICY) return undefined
```

Comment:
suggestion: let's update the format a bit as it's very hard to parse as a reader. I suggest the following.


```suggestion
    const adapterNode = nodes.find((node) => node.id === selectedNode) as Node<TransitionData> | undefined
    if (!adapterNode) {
      return undefined
    }
    
    const incomers = getIncomers(adapterNode, nodes, edges)
    if (!incomers || incomers.length !== 1) {
      return undefined
    }

    const [behavior] = incomers
    if (behavior.type !== DataHubNodeType.BEHAVIOR_POLICY) {
      return undefined
    }
```

Reply:
Nice one. ESLint's `curly` rule is not set. Will add it asap.